### ISSUE TITLE
TypeError using validate=False, Parser.DataHandler.directory

### ISSUE THREAD

**Issue by 6syun9** [biopython/biopython#2048]
### Setup

```
3.6.8 (default, Apr  9 2019, 17:16:47)
[GCC 4.2.1 Compatible Apple LLVM 10.0.0 (clang-1000.11.45.5)]
CPython
Darwin-17.7.0-x86_64-i386-64bit
1.73
```

### Expected behaviour

I can fetch pubmed using `/tmp` directory when validation param is False : `Entrez.read(handle,validate=False)`
In AWS Lambda, we need to make this function.

### Actual behaviour

```
Traceback (most recent call last):
  File "pubmed_downloader.py", line 69, in _get_json
    record = handler.read(handle)
  File "~~~/site-packages/Bio/Entrez/Parser.py", line 368, in read
    self.parser.ParseFile(handle)
  File "/private/var/folders/xj/m19b8sv15cqfhvrsz1w0bl3w0000gp/T/python-build.20190409171539.50338/Python-3.6.8/Modules/pyexpat.c", line 473, in EndElement
  File "~~~/site-packages/Bio/Entrez/Parser.py", line 559, in endElementHandler
    consumed = consumer.endElementHandler(name, prefix)
TypeError: endElementHandler() takes 2 positional arguments but 3 were given
```

### Steps to reproduce

```
from Bio import Entrez
from Bio.Entrez import Parser

with Entrez.efetch(db="pubmed", id="29963580", retmode="xml") as handle:
    handler = Parser.DataHandler(False, False)
    handler.directory = '/tmp'
    record = handler.read(handle)
```

**Comment by 6syun9** [comment]
I examined the cause, this issue had happen by `pubmed, id=29963580` only. (I tried 50,000 files from random sampling)

**Comment by peterjc** [comment]
@mdehoon do you want to look into this corner case? It might be an NCBI bug...

**Comment by mdehoon** [comment]
It's a bug in our parser.

**Comment by 6syun9** [comment]
Isn't it good to do add `mprescripts` in the DTD?

> handler = Parser.DataHandler(True, False)
`ValidationError: Failed to find tag 'mprescripts' in the DTD. To skip all tags that are not represented in the DTD, please call Bio.Entrez.read or Bio.Entrez.parse with validate=False.`

**Comment by peterjc** [comment]
Reopened as Michiel said its a bug in our parser.

@6syun9 we don't want to edit the DTD files - the NCBI creates them to programatically describe what should be in their XML files. There have occasionally been mismatches, which we have reported to the NCBI to have fixed.

**Comment by 6syun9** [comment]
@peterjc  Thanks. I understand.
> we don't want to edit the DTD files

**Comment by mdehoon** [comment]
@6syun9  actually it is present in the DTD.

**Comment by mdehoon** [comment]
Fixed by #2056
